### ISSUE TITLE
Directory issues -- non-relative even with ../db.

### ISSUE THREAD

**Issue by okram** [thinkaurelius/titan#423]
Please see the last line of the load. In the conf/ you have ../db. Notice that it is not respected as relative. 

```
~/software/aurelius/titan/conf$ more titan-hazelcast.properties
storage.backend=hazelcastcache
storage.directory=../db
```

I would make all the temp directories /tmp to be safe.

```
gremlin> g = TitanFactory.open('conf/titan-hazelcast.properties')
Oct 14, 2013 8:24:47 AM com.hazelcast.config.XmlConfigBuilder
INFO: Looking for hazelcast.xml config file in classpath.
Oct 14, 2013 8:24:47 AM com.hazelcast.config.XmlConfigBuilder
WARNING: Could not find hazelcast.xml in classpath.
Hazelcast will use hazelcast-default.xml config file in jar.
Oct 14, 2013 8:24:47 AM com.hazelcast.config.XmlConfigBuilder
INFO: Using configuration file file:/Users/marko/software/aurelius/titan/lib/hazelcast-all-3.0.2.jar!/hazelcast-default.xml in the classpath.
Oct 14, 2013 8:24:48 AM com.hazelcast.instance.DefaultAddressPicker
INFO: Prefer IPv4 stack is true.
Oct 14, 2013 8:24:48 AM com.hazelcast.instance.DefaultAddressPicker
INFO: Picked Address[192.168.1.124]:5701, using socket ServerSocket[addr=/0:0:0:0:0:0:0:0,localport=5701], bind any local is true
Oct 14, 2013 8:24:48 AM com.hazelcast.system
INFO: [192.168.1.124]:5701 [dev] Hazelcast Community Edition 3.0.2 (20130906) starting at Address[192.168.1.124]:5701
Oct 14, 2013 8:24:48 AM com.hazelcast.system
INFO: [192.168.1.124]:5701 [dev] Copyright (C) 2008-2013 Hazelcast.com
Oct 14, 2013 8:24:48 AM com.hazelcast.instance.Node
INFO: [192.168.1.124]:5701 [dev] Creating MulticastJoiner
Oct 14, 2013 8:24:48 AM com.hazelcast.core.LifecycleService
INFO: [192.168.1.124]:5701 [dev] Address[192.168.1.124]:5701 is STARTING
Oct 14, 2013 8:24:52 AM com.hazelcast.cluster.MulticastJoiner
INFO: [192.168.1.124]:5701 [dev]


Members [1] {
    Member [192.168.1.124]:5701 this
}

Oct 14, 2013 8:24:52 AM com.hazelcast.core.LifecycleService
INFO: [192.168.1.124]:5701 [dev] Address[192.168.1.124]:5701 is STARTED
WARN  com.thinkaurelius.titan.diskstorage.hazelcast.AbstractHazelcastStoreManager  - Hazelcast does not support multiple transactions per thread
==>titangraph[hazelcastcache:/Users/marko/software/aurelius/titan/conf/../db]
```

**Comment by dalaro** [comment]
Regarding hazelcast.xml: I recently fixed this in the hazelcast distro but missed it in the all/standalone distro.  Fixed with (69a156e0249a0173b717cdade73db96a53321bcb).

Regarding relative paths: I don't understand.  The last line shows interrpretation of ../db relative to the directory containing the config file in which it appeared, which is the new intended behavior for relative paths.  When I set `storage.directory = /tmp/hc` and restarted, I got `==>titangraph[hazelcastcache:/tmp/hc]`.  This is the unchanged intended behavior for absolute paths.

Are you saying we should drop the storage directory entirely for hazelcastcache?  AFAIK we don't support Hazelcast's map persistence, so I suspect there is no reason for this directory to exist.  Maybe I am mistaken.

**Comment by okram** [comment]
Yes, it is relative, apologies. I just noticed it. However, I think we should use /tmp/db as opposed to the relative path ../  (but that is just a subjective preference). Either works --- though /Users/marko/blah/blam/bleep/../db is a weird looking toString() of g.

**Comment by dalaro** [comment]
/tmp is best for totally impermanent data.  Not all systems run [tmpwatch](https://fedorahosted.org/tmpwatch/) or delete the contents of /tmp every time the system reboots, but many do.  For this reason, I'm nervous about writing /tmp into our default configs.  /var/lib/titan is probably the optimal LSB path when root privileges are available, but we shouldn't require root for the distro zipfiles.  So I went with this relative path as a compromise.  Its a directory inside the git repo or the extracted zipfile directory -- something guaranteed to be writable and unlikely to be swept away by aggressive /tmp cleaning policies.

Anyway, that was my rationale.  I'm certainly open to better ideas.

**Comment by dalaro** [comment]
Ah, regarding dropping the directory, #420 might be related...

**Comment by okram** [comment]
The properties files are for demo purposes. I would use /tmp for this. When someone wants to get "serious" they can create their own properties file. But for intial playing and having tutorials, this just looks rediculous:

```
==>titangraph[hazelcastcache:/Users/marko/software/aurelius/titan/conf/../db]
```

I would do this as the general pattern:

```
/tmp/titan-hazelcast
/tmp/titan-embeddedcassandra
```

This way the g.toString() looks good and yes, shows people: "be sure to update this when you get serious about your data beyond the default properties provided."

**Comment by dalaro** [comment]
Is this primarily a concern about toString()?  If so, then would you accept a toString() based on the current directory, such that after executing `bin/gremlin.sh` you would see `==>titangraph[hazelcastcache:db]` and if you instead changed to the `bin` directory and ran `./gremlin.sh` then you would see `==>titangraph[hazelcastcache:../db]`?

**Comment by dalaro** [comment]
Alternately, the samples above could be prefixed with a "./" if you want to emphasize that they are both filesystem paths.

**Comment by mbroecheler** [comment]
My two cents: Many people use the default configuration files. They may copy them to a different directory (or not even) and then start going to town with them. Leaving them at /tmp/ is dangerous because of the potential loss of data.
For this reason, Cassandra uses /var/ for their data directories, but that sucks too because of permissions and so you either need to grant permissions or change the default yaml file.

I think the relative directory thing is a good middle of the road approach. It is reasonably safe and the toString() issue is simply cosmetic.
Bottom line: defaults rule the world so we got to be careful.

**Comment by white127** [comment]
Can I modify hazelcast conf, for example, I want to use the tcp-ip method to join cluster, not the default multicast
